### PR TITLE
improve the flutter error ids we generate for analytics

### PR DESCRIPTION
**Pull request by devoncarew**
- improve the flutter error ids we generate for analytics
- fix https://github.com/flutter/flutter-intellij/issues/4467

This fixes the `'package` error ID we'd seen in analytics.

### CODE REVIEW

File: testSrc/unit/io/flutter/logging/FlutterErrorHelperTest.java [hunk 1]
```diff
@@ -34,27 +34,84 @@ public void testGetAnalyticsId_scrubbing() {
...
       FlutterErrorHelper.getAnalyticsId("could not find a generator for route routesettings(\"/foobar\", null) in the _widgetsappstate"));
     assertEquals(
-      "controller's-length-property-does-not-match-the",
-      FlutterErrorHelper.getAnalyticsId("controller's length property (123) does not match the\nnumber of tabs"));
+      "controller's-length-property-does-not-match-the-number-of-tabs",
+      FlutterErrorHelper.getAnalyticsId("controller's length property (123) does not match the number of tabs"));
+  }
+
+  @Test
+  public void testGetAnalyticsId_errorStudiesApp() {
```
Comment (devoncarew) [inline testSrc/unit/io/flutter/logging/FlutterErrorHelperTest.java:66]: @InMatrix - can you verify that the IDs here are roughly what we'd want to see for the given errors?

Comment (InMatrix) [reply testSrc/unit/io/flutter/logging/FlutterErrorHelperTest.java:66]: Yes, those IDs look good. I'm not sure if we need to include the package information though. Is it included to disambiguate similar assertions?

Comment (devoncarew) [reply testSrc/unit/io/flutter/logging/FlutterErrorHelperTest.java:66]: By the package info, do you mean the `package:flutter_error_studies/...` text in the source errors below? It's just the text that I copied in - it's nice to have real world examples (definitely an issue when I originally wrote this, to have to guess what the error might look like).

Comment (InMatrix) [reply testSrc/unit/io/flutter/logging/FlutterErrorHelperTest.java:66]: No, I was referring to "'package:flutter/src/material/reorderable_list.dart': failed assertion: line 123 pos 123: 'children.every((widget w) => w.key != null)'"));" under "all children of this widget must have a key.\n" at line 56. I'm not sure if the extra detail about the specific file in package flutter and assertion position in the file are necessary for analytics .

**Review by stevemessick — APPROVED** [review]

**Pull request merged**